modules/launcher/src/main/java/org/mule/runtime/module/launcher/application/DefaultApplicationFactory.java

```diff
@@ -82,7 +96,9 @@ protected Application createAppFrom(ApplicationDescriptor descriptor) throws IOE
 
         parent = getSharedLibClassLoader(descriptor, parent);
 
-        final List<ApplicationPlugin> applicationPlugins = createApplicationPlugins(parent, descriptor.getPlugins());
+        final List<ApplicationPlugin> applicationPlugins = createContainerApplicationPlugins(parent, descriptor);
```

Comment:
Maybe these dependencies can be injected in order to make the class more testable